mlflow/metrics/genai/genai_metric.py

```diff
@@ -245,6 +246,14 @@ def make_genai_metric_from_prompt(
         )
 
     """
+    # DO NOT INSERT ANY LOGIC ABOVE. We want locals() to capture initial arguments only.
+    # When users create a custom metric using this function,the metric configuration
+    # will be serialized and stored. This enables us to later deserialize the configuration,
+    # allowing users to understand their LLM evaluation results more clearly.
+    genai_metric_args = locals()
+    # Record the mlflow version for serialization in case the function signature changes later
+    genai_metric_args["mlflow_version"] = VERSION
```

Comment:
I just saw this note:

<img width="834" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/182709ff-5af7-49db-b30a-a8b538a9878f">

>  The contents of this dictionary should not be modified; changes may not affect the values of local and free variables used by the interpreter.

Reply:
oh got it. what's the right way to understand this note? my interpretation is that

```
a = 10
locs = locals()
locs["a"] = 30

print(a) # a should still be 10 not 30 cuz locs["a"] = 30 just changes the snapshot, not affect the actual values of locals.
```

and this seems to be aligned with what we want in this case.

Reply:
can you create a copy or, construct the dictionary manually?

Reply:
discussed offline, will move to manual creation